internal/nginx/modules/httpmatches.js

```diff
@@ -0,0 +1,83 @@
+export default { redirect };
+
+const matchesVariable = 'http_matches';
+
+// FIXME(osborn): Need to add special handling for repeated headers.
+// Should follow guidance from https://www.rfc-editor.org/rfc/rfc7230.html#section-3.2.2.
+function headersMatch(r, headers) {
+  for (let i = 0; i < headers.length; i++) {
+    const h = headers[i];
+    const kv = h.split(':');
+    if (kv.length !== 2) {
+      r.error('invalid header match: ' + h);
```

Comment:
I wonder if in this case it is more appropriate to return 500 ( Internal Server Error) as the reason will be a bug in the go code. 

same comment for paramsMatch

Reply:
Refactored the njs code a bit to address this comment. We will now return 500 for all malformed matches.